### PR TITLE
Changes to `cocotb.log`

### PR DESCRIPTION
**Pull request by ktbarrett**
Removes the superfluous `SimLog` function and make the `cocotb.log` module private.

### CODE REVIEW

**Comment by cmarqu** [comment]
https://github.com/cocotb/cocotb/blob/master/docs/source/rotating_logger.rst needs adaptations as well.
Ideally we would run the code as a test (like in https://github.com/cocotb/cocotb/blob/master/docs/source/quickstart.rst?plain=1#L41-L44) and only include (parts of) the file in the docs.

File: src/cocotb/scheduler.py [hunk 1]
```diff
@@ -103,7 +102,9 @@ def __init__(self):
         self.event = Event()
         self.state = external_state.INIT
         self.cond = threading.Condition()
-        self._log = SimLog(f"cocotb.external.thead.{self.thread}", id(self))
+        self._log = logging.getLogger(
+            f"cocotb.external.thead.{self.thread}.0x{id(self):x}"
```
Comment (cmarqu) [inline src/cocotb/scheduler.py:106]:
```suggestion
            f"cocotb.external.thread.{self.thread}.0x{id(self):x}"
```

**Comment by ktbarrett** [comment]
Hmmm... okay. I guess we are providing the classes in `cocotb.log` as public interfaces. I guess this should stay public. Though I *really* don't like shadowing the module `cocotb.log` with the variable `cocotb.log`, so I might change the name to `cocotb.logging` instead.

**Comment by ktbarrett** [comment]
Odd...

Apparently, importing an object from a module also does an import of that module. And Python also fixes up relative modules `import cocotb.logging` into relative imports `import .logging` which brings in the name `logging`. So in cocotb/__init__.py when doing `from cocotb.logging import ...` the `logging` name is overwritten which was once the standard library logging module. Very very strange behavior IMO.

**Comment by cmarqu** [comment]
So the least surprising way would be to call it something like `cocolog`?

**Comment by ktbarrett** [comment]
Well it only matters in the implementation of this file and specifically only on the call to `logging.getLogger` in `__init__.py`. It's not an issue for the user.

**Review by imphil — APPROVED** [review]

**Pull request merged**